numba/tests/test_typedlist.py

```diff
@@ -101,6 +107,76 @@ def test_basic(self):
         # index
         self.assertEqual(l.index(15), 4)
 
+    def test_unsigned_access(self):
+        L = List.empty_list(int32)
+        ui32_0 = types.uint32(0)
+        ui32_1 = types.uint32(1)
+        ui32_2 = types.uint32(2)
+
+        # insert
+        L.append(types.uint32(10))
+        L.append(types.uint32(11))
+        L.append(types.uint32(12))
+        self.assertEqual(len(L), 3)
+
+        # getitem
+        self.assertEqual(L[ui32_0], 10)
+        self.assertEqual(L[ui32_1], 11)
+        self.assertEqual(L[ui32_2], 12)
+
+        # setitem
+        L[ui32_0] = 123
+        L[ui32_1] = 456
+        L[ui32_2] = 789
+        self.assertEqual(L[ui32_0], 123)
+        self.assertEqual(L[ui32_1], 456)
+        self.assertEqual(L[ui32_2], 789)
+
+        # index
+        ui32_123 = types.uint32(123)
+        ui32_456 = types.uint32(456)
+        ui32_789 = types.uint32(789)
+        self.assertEqual(L.index(ui32_123), 0)
+        self.assertEqual(L.index(ui32_456), 1)
+        self.assertEqual(L.index(ui32_789), 2)
+
+        # delitem
+        L.__delitem__(ui32_2)
+        del L[ui32_1]
+        self.assertEqual(len(L), 1)
+        self.assertEqual(L[ui32_0], 123)
+
+        # pop
+        L.append(2)
+        L.append(3)
+        L.append(4)
+        self.assertEqual(len(L), 4)
+        self.assertEqual(L.pop(), 4)
+        self.assertEqual(L.pop(ui32_2), 3)
+        self.assertEqual(L.pop(ui32_1), 2)
+        self.assertEqual(L.pop(ui32_0), 123)
+
+    # @unittest.skipIf(config.IS_32BITS,
+    #                  "prange is not supported on 32 bit hardware")
+    # @unittest.skipIf(config.IS_WIN32 and (not IS_PY3),
+    #                  "parallel target is not supported on Windows with PY2")
+    @unittest.skipIf(not (_windows_py27 or _32bit),
+                     "prange not supported on windows w/ PY2 or 32 bit hardware")
```

Comment:
This probably needs `from .test_parfors import skip_unsupported as parfors_skip_unsupported` adding in the imports section at the top, and then decorate this test with that. The above can all be removed.